File: csharp/lesson 6/InOrder.cs
// https://leetcode.com/problems/binary-tree-inorder-traversal/

class Solution 
{
    public IList<int> InorderTraversal(TreeNode root)
    {
        var result = new List<int>();
        InOrder(root, result);
        return result;
    }

    private void InOrder(TreeNode node, List<int> result)
    {
        if (node == null) return;

        InOrder(node.left, result);
        result.Add(node.val);
        InOrder(node.right, result);
    }
}

File: ruby/lesson 6/in_order.rb
# https://leetcode.com/problems/binary-tree-inorder-traversal/

def inorder_traversal(root)
    result = []
    inorder(root, result)
    result
end

def inorder(node, result)
    return if node.nil?

    inorder(node.left, result)
    result.push(node.val)
    inorder(node.right, result)
end
